test/runtests.jl

```diff
@@ -194,6 +194,60 @@ end
         @inferred constructorof(typeof(lr1))(getfields(lr2)...)
     end
 
+    @testset "Cholesky" begin
+        X = Matrix(Diagonal(ones(3)))
+        @testset "uplo=$uplo" for uplo in ['L', 'U']
+            C = Cholesky(X, uplo, 0)
+
+            @testset "only setting value" begin
```

Comment:
It seems these test would also pass if `setproperties` was a noop?

Reply:
Maybe add some tests with some explicit 2x2 matrices?

Reply:
Is this better?

Reply:
I had in mind explicit matrix literals like `[1 0; 0 2]`, but this is also fine.